app/facades/results_facade.rb

```diff
@@ -0,0 +1,12 @@
+class ResultsFacade
+  def self.results(q)
+    @foods = FoodService.new.results(q).first(10).map do |data|
+        Food.new(data)
+      end
```

Comment:
This is great. We want to take the first 10 from the results and then create our Food poros with that data. No need for the `@foods` instance variable though. Instance variables are only passed from controllers to views. The return of your `map` which is the array of Food objects is returned to your controller action that calls this method.